decentralized_identity_proof_DNS-TXT.md

```diff
@@ -91,6 +91,33 @@ This will unfortunately be an unresolved issue if we use DNS, as DNS has no arch
 #### DNS Spoofing
 [DNS spoofing](https://en.wikipedia.org/wiki/DNS_spoofing) is partially mitigated by using a well known DNS resolver such as Google or Cloudflare to retrieve the TXT records. However using DNSSEC can fully resolve the issue.
 
+## Using DNS and DID to verify the identity of a signer
+When signing a certificate, one can add it's DID into the `proof.verificationMethod` field, so that OA can automatically verify that the DID was the one that signed the certificate:
+```json
+proof: {
+  type: "EcdsaSecp256k1Signature2019",
+  created: "2020-04-26T21:41:34.663Z",
+  proofPurpose: "assertionMethod",
+  verificationMethod: "did:ethr:ropsten:0x44E682d207bcDDDAD0Bb3a650cCb9de0911B9D3A#owner",
+  signature:      "0x49898c7ded0bef0e3fb96b3f69b097dda45a474c62142d72a0834d814c092cbe458a16f44efe136614e74ffd69b8273688af347826b9efaccd6a12f200185eef1c"
```

Comment:
```suggestion
proof: {
  type: "DID",
  created: "2020-04-26T21:41:34.663Z",
  proofPurpose: "assertionMethod",
  did: "did:ethr:ropsten:0x44E682d207bcDDDAD0Bb3a650cCb9de0911B9D3A#owner",
  signature:      "0x49898c7ded0bef0e3fb96b3f69b097dda45a474c62142d72a0834d814c092cbe458a16f44efe136614e74ffd69b8273688af347826b9efaccd6a12f200185eef1c"
```

suggested the above change because we really dont know what type of signature scheme the DID is using at the point of reading this (assuming you know nothing about `ethr` method).

So I imagine:
1. I see `DID type`, I look up the `did` field to see the did to resolve to (did:ethr:ropsten:0x44E682d207bcDDDAD0Bb3a650cCb9de0911B9D3A)
2. I do a lookup on universal resolver to see that did:ethr:ropsten:0x44E682d207bcDDDAD0Bb3a650cCb9de0911B9D3A resolve to many public key, one of it is 0x44E682d207bcDDDAD0Bb3a650cCb9de0911B9D3A and the method is `EcdsaSecp256k1Signature2019`
3. I will then use the `EcdsaSecp256k1Signature2019` method to verify `signature` with the public key `0x44E682d207bcDDDAD0Bb3a650cCb9de0911B9D3A`.

Reply:
@AdamJLemmon

Reply:
These proofs are actually not dependent on DIDs at all. A DID *may* just be used to identify the `verificationMethod`.

So the process for the verifier would go something like this:

1. Check if you support the proof type presented: `EcdsaSecp256k1Signature2019`
2. If supported: look up the verification method. This is most commonly a URL to a `publicKey` but in our current implementation is just the publicKey itself. This is the only place where DID support would be required, to resolve the URL(DID).  You then resolve the URL and retrieve the publicKey.
3. The verifier than verifies `proof.signature` against the resolved `verificationMethod` using a method determined by `proof.type`.

[This](https://github.com/Open-Attestation/oa-verify/blob/7cb7499bab4ba4707ed5b6b5b770abc0b42be0f8/src/verifiers/signedProof/openAttestationSignedProof.ts#L31) is actually the complete implementation in `oa-verify` to verify these proof blocks without DIDs at present.

I would prefer to keep this loosely coupled with DIDs and allow for flexibility of proof types as well as verification methods. Ie Ed25519 sigs with my publicKey hosted on github :)